### DISCUSSION TITLE
Welcome to Advanced Python Mastery Discussions!

### DISCUSSION THREAD

**Discussion by dabeaz** [dabeaz-course/python-mastery#1] [Announcements]
## Welcome!
  Use discussions to:
  * Ask questions about the course
  * Share ideas and interesting resources. 
  * Engage with other members who are working on the course material.

I'm not always available to answer detailed course questions, but perhaps discussions can be used as an ongoing knowledge base for the course.

Cheers,
Dave

**Comment by AbdSuperDev** [comment]
Hi, I am happy to join us

**Comment by BtAbdul** [comment]
this course available in videos lectures?

**Reply by dabeaz** [reply]
As noted in the README, there are no video lectures for the course.    It's not something that I would be likely to produce due to the time involved and the fact that I already have a closely related video series on Safari (https://www.oreilly.com/library/view/python-programming-language/9780134217314/).

**Reply by BtAbdul** [reply]
the course is valuable, thank you so much sir ❤

**Comment by AbdSuperDev** [comment]
thank you, i'm studying data science and AI, and i want to master python, is this course for me?

**Reply by dabeaz** [reply]
This course won't say much about data science and AI (specifically as topics), but the course was frequently taken by scientists, analysts, and other people involved with using Python for data analysis.   A lot of them were looking to move beyond writing simple scripts (or notebooks) to more complete programs.  Also, the course aims to introduce many of the programming techniques that get used in libraries and frameworks--which could be of interest if you're in a position of having to read other people's code.

**Reply by AbdSuperDev** [reply]
Ok, I understood, thanks for explain, I hope learn a lot of knowledge of python with you !
thank's

**Comment by CharlieYu1** [comment]
Question about Exercise 3.5. Is it possible to make create_formatter as a function of TableFormatter class? So that it takes a string and returns the appropriate subclass
